addon/components/frost-checkbox.js

```diff
@@ -76,18 +67,59 @@ export default Component.extend(PropTypeMixin, {
     }
   },
 
-  // == Events ================================================================
+  // == DOM Events ============================================================
 
+  /**
+   * Handle the focusIn event
+   * @params {Event} e - the event being handled
+   */
   _onFocus: Ember.on('focusIn', function (e) {
     // If an onFocus handler is defined, call it
     if (this.attrs.onFocus) {
       this.attrs.onFocus()
     }
   }),
 
+  /**
+   * Handle the keypress event
+   * @param {Event} e - the event being handled
+   * @returns {Boolean|undefined} false if we want to stop propagation
+   */
+  keyPress (e) {
+    if (e.keyCode === 32) {
+      if (get(this, 'disabled') !== true) {
+        this.$('input').prop('checked', !this.$('input').prop('checked'))
+        this.send('input')
+      }
+      e.preventDefault()
+      e.stopPropagation()
+      return false
+    }
+  },
+
+  // == Lifecycle Hooks =======================================================
+
+  /* Ember.Component method */
```

Comment:
I commented on a similar case above, but this is a good example of what I'm talking about - if it's unclear that everything inside of Lifecycle Hooks is from the component, then we might as well call the header `Ember Component Lifecycle Hooks` and do away with the (in my opinion) redundant jsdocs